src/pgwire/src/protocol.rs

```diff
@@ -695,7 +701,16 @@ where
             // statement.
             self.start_transaction(Some(num_stmts));
 
-            match self.one_query(stmt).await? {
+            let kind = StatementKind::from(&stmt);
+
+            let start = Instant::now();
+            let result = self.one_query(stmt).await;
+            let latency = start.elapsed();
+
+            self.metrics
+                .one_query_latency(result.is_ok(), kind, latency);
```

Comment:
This doesn't measure queries using the extended protocol, which many drivers use (see `fn execute` in this file).

The large result set or slow client network connection I think makes this misleading. Although it's some annoying work, can this get changed to set the metric in the `send_rows` function at the time it sees its first row? Time from query received to first row internally feels most useful to be.

Reply:
ah good point. put up a "time to first row" metric. the code's a little gnarly, especially around what kinds of queries/query results we count in this metric. Right now it's all select/subscribes/copy that return rows (so no error return in parse/bind/etc or during query execution and no cancellations)